ext/ffi/lib.rs

```diff
@@ -1077,6 +1189,19 @@ where
       NativeType::Pointer | NativeType::Function => NativeValue {
         pointer: cif.call::<*const u8>(*fun_ptr, &call_args),
       },
+      NativeType::Struct(fields) => {
+        let size = NativeType::compute_struct_size(fields);
```

Comment:
Optionally, the struct size could be given as the value for `NativeType::Struct()` in the parameter / return type to remove the need to compute the struct size over and over again.

This would probably need an op to compute the size, allowing users to easily calculate size of structs that depend on the system pointer size.

Reply:
It's only calculated once and that's for return value

Reply:
Is the parameter structure declaration even relevant then? Does libffi use the structure contents somehow?

Also, do we want to check the buffer size to match the structure?

Reply:
> Also, do we want to check the buffer size to match the structure?

I don't think it's worth it